### ISSUE TITLE
Could we use STAR for mapping?

### ISSUE THREAD

**Issue by dinktnwo** [dincarnato/RNAFramework#27]
Hi RNAFramework teams,
I'm a newbie in the field of RNA. You did a great work for whole transcritome SHAPE analysis. However, there are some questions bug me a lot. 
1. You are using bowtie2 for mapping and mutation-counts, could we change to STAR? it seems faster than bowtie2 in terms of whole transcriptome. 
2. There are some alternative junctions between different transcripts in whole transcriptome, would they cause the unwanted mutation counts? For example, for seq, AAACT, AAACTA, would the last A would count as mutation in the first seq?
Thanks.

**Comment by dincarnato** [comment]
Hi dinktnwo,

so:

1. You can use any mapper you want to perform the alignment of reads to the transcriptome. Feel free to choose STAR, BWA, or anything you like. Simply perform the alignment manually, skip the rf-map step, and go straight to the rf-count step with the SAM/BAM file you generated.
2. I am not sure I understand the question here. Can you rephrase it, or maybe provide a more detailed example?

Best,
Danny

**Comment by dinktnwo** [comment]
Hi Danny,

Thanks for such quickly reply and sorry for my un-clear question. 
My second question is about the count of mutations or RT-stops. I have no idea the results of rf-count would change or not if I choose STAR or other mapping program. 
For example, if my reference fasta is AAAAA and bowtie2 mapping with AAAAT, then it would count the T as 1 mutation. And if there is a junction at AAAAA, let's say AAAA|A (not real), Bowtie2 would map AAAA- and AAAAT to AAAAA, and count the mutation twice. However if i use STAR, it may think AAAA- is another transcript (which I didn't give to bowtie2), and only count one mutation (AAAAT).

Thanks,
dinktnwo

**Comment by dincarnato** [comment]
Hi dinktnwo,

this is too hypothetical and I don't have a binary answer for you.
What I can say is that, on the full experiment, the individual bias of either of the two aligners should impact only minimally the final results.
Something worth mentioning is that, when dealing with MaP data, you should avoid end-to-end mapping of reads, but enable soft-clipping. This way, terminal mutations (in your example AAAAT read vs. AAAAA ref) will be clipped and ignored.

Best,
Danny

**Comment by dinktnwo** [comment]
Hi Danny,

Thanks for your kindly reply and recommendation. It's very very helpful ! I think it's all my questions.

Best,
dinktnwo
